logstash-core-event-java/src/main/java/com/logstash/ext/JrubyEventExtLibrary.java

```diff
@@ -251,9 +251,24 @@ public IRubyObject ruby_to_java(ThreadContext context)
 
         @JRubyMethod(name = "to_json", rest = true)
         public IRubyObject ruby_to_json(ThreadContext context, IRubyObject[] args)
-            throws IOException
         {
-            return RubyString.newString(context.runtime, event.toJson());
+            try {
+                return RubyString.newString(context.runtime, event.toJson());
+            } catch (Exception e) {
+                throw new RaiseException(context.runtime, context.runtime.getModule("LogStash").defineOrGetModuleUnder("Json").getClass("GeneratorError"), e.getMessage(), true);
```

Comment:
this usage of `defineOrGetModuleUnder` is slightly confusing.

If you end up defining it, then wouldn't the proceeding `getClass` return null? since nothing is using it after that fact, I suppose it is fine